controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -136,9 +138,11 @@ func (w *Workload) UpdateCoreDNS(ctx context.Context, kcp *controlplanev1.Kubead
 		return nil
 	}
 
-	// Validate the image tag.
-	if err := validateCoreDNSImageTag(info.FromImageTag, info.ToImageTag); err != nil {
-		return errors.Wrapf(err, "failed to validate CoreDNS")
+	// Validate the image tag if it was changed.
+	if info.FromImageTag != info.ToImageTag {
```

Comment:
required due to https://github.com/kubernetes-sigs/cluster-api/pull/6917/files#r922006034 .

Should we prevent updating if the image tag does not change?

Reply:
The issue was:

if `FromImageTag == ToImageTag`, `validateCoreDNSImageTag` would have returned an error which would result in not updating coredns and failed reconciliation.

Reply:
I moved the change to the `validateCoreDNSImageTag` func.

If we not allow keeping the image tag the same, we don't allow to change the image repository.